### PR TITLE
Lovelace Drag and Drop + Resize

### PR DESCRIPTION
**Pull request by zsarnett**
## Breaking change

Probably

## Proposed change

New Layout system - Drag and Drap using Lit Grid Layout

66e2b1ea0199383b80dc6dc72c7bcc88

## Type of change

- [x] New feature (thank you!)

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

Notes to self for todo: 

- [ ] Create Documentation somewhere on how to use this feature
- [x] Instead of the current Card Options Footer appended to the card. Add hover feature similar to Hangouts
- [x] Rework for the New Custom View Feature
- [ ] Release as a Custom View before the 0.117 (Depending on the timing of review and finished v1 product)

### CODE REVIEW

File: src/panels/lovelace/views/hui-view.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ import {
   css,
 } from "lit-element";
 import { classMap } from "lit-html/directives/class-map";
+import "lit-grid-layout";
```
Comment (bramkragten) [inline src/panels/lovelace/views/hui-view.ts:12]: Should be loading in edit code

File: src/panels/lovelace/views/hui-view.ts [hunk 2]
```diff
@@ -101,9 +91,17 @@ export class HUIView extends LitElement {
...
+    console.log("render");
+
     return html`
       <div id="badges"></div>
-      <div id="columns"></div>
+      <lit-grid-layout
+        .dragHandle=${"hui-card-options"}
+        .items=${this._cards}
+        .layout=${this._layout}
+        .rowHeight=${15}
```
Comment (bramkragten) [inline src/panels/lovelace/views/hui-view.ts:102]: Assuming this property has type number in `lit-grid-layout`
```suggestion
        rowHeight="15"
```

**Comment by zsarnett** [comment]
I think I want to be able to save the Layout Config Array on the View itself instead of giving the information to the card. But when I save the lovelace config of the view the view is recreated each time a card is moved. Any suggestions on how to save the layout to the view?

**Comment by zsarnett** [comment]
I need a way to save Lovelace View config without triggering the view to be recreated in the root. I need to save it when a user moves a card as well as on first run of the view when the view calculates the layout based on the cards and the already save layout (new card added in YAML, it needs to position that card some where and then save it)

**Comment by XtremeOwnageDotCom** [comment]
That is disappointing :-(

**Comment by Pirol62** [comment]
So sad. But I can understand how difficult it is, to store the settings in a performant and secure way ecen during the edit mode.

**Comment by lzinga** [comment]
Is there any kind of update with this?

**Pull request closed without merging**